lib/mocha/mocha.ts

```diff
@@ -0,0 +1,93 @@
+'use strict';
+
+((context) => {
+  var Mocha = context['Mocha'];
+  
+  if (typeof Mocha === 'undefined') {
+    throw new Error('Missing Mocha.js');
+  }
+
+  if (typeof Zone === 'undefined') {
+    throw new Error('Missing Zone.js');
+  }
+
+  let ProxyZoneSpec = Zone['ProxyZoneSpec'];
+
+  if (!ProxyZoneSpec) {
+    throw new Error('Missing ProxyZoneSpec');
+  }
+
+  if(Mocha['__zone_patch__']){
+    throw new Error('"Mocha" has already been patched with "Zone".');
+  }
+
+  Mocha['__zone_patch__'] = true;
+
+  let rootZone = Zone.current;
+
+  function wrapCallbackInProxyZone(fn: Function): Function {
```

Comment:
I don't think it is that simple. We need to have one proxy zone for the whole test not a separate instance per callback. Even if the tests pass, this can't be right.

Turns out that I have already started work on `Mocha` integration, but have not finished it. Do you think you could merge my work with yours and we could get it in that way?

https://github.com/mhevery/zone.js/commit/ec0f057d350bdd3e8e78104b3b788df2c6d1feed